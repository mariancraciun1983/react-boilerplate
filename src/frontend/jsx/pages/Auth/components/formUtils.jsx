/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import className from 'classnames';

const getComponentToRender = (type, params = {}) => {
  if (type === 'textarea') {
    return <textarea {...params} />;
  }
  return <input {...params} type={type} />;
};

export const renderField = (dt) => {
  const {
    input,
    autoComplete,
    label,
    type,
    meta: { touched, error },
  } = dt;
  const hasError = touched && error;
  const hasSuccess = touched && !error;
  const params = {
    ...input,
    className: className({ 'form-control': true, 'is-valid': hasSuccess, 'is-invalid': hasError }),
    placeholder: label,
  };
  if (autoComplete) params.autoComplete = autoComplete;
  return (
    <div>
      <div className="form-group">
        {label ? <label htmlFor={params.name}>{label}</label> : null}
        {getComponentToRender(type, params)}
        {hasError ? <div className="invalid-feedback">{error}</div> : null}
      </div>
    </div>
  );
};

export const onSubmitFail = (errors) => {
  const key = Object.keys(errors)
    .reduce((k, l) => {
      if (document.getElementsByName(k)[0].offsetTop < document.getElementsByName(l)[0].offsetTop) {
        return k;
      }
      return l;
    });
  if (document.getEleentsByName(key).length) {
    document.getElementsByName(key)[0].focus();
  }
};
