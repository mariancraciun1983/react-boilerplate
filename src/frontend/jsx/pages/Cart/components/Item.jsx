/* eslint-disable radix */
import React from 'react';
import PropTypes from 'prop-types';
import { formatPrice } from '../../../../core/utils/formatter';

class Item extends React.PureComponent {
  constructor(props) {
    super(props);
    this.onUpdateMovie = this.onUpdateMovie.bind(this);
    this.onDelete = this.onDelete.bind(this);
  }

  onUpdateMovie(e) {
    const { onUpdate, movie: { id } } = this.props;
    onUpdate(id, parseInt(e.target.value));
  }

  onDelete() {
    const { onUpdate, movie: { id } } = this.props;
    onUpdate(id, 0);
  }

  render() {
    const { quantity, movie } = this.props;
    return (
      <div className="d-flex mb-3 pb-3 border-bottom">
        <div className="mr-auto d-inline-flex">
          <img src={movie.image} alt="Movie" />
          <div className="p-3">
            <strong>{movie.title}</strong>
            <br />
            <span>{`Price: ${formatPrice(movie.price)}$`}</span>
          </div>
        </div>
        <div className="">
          Quantity:
          <input
            className="form-control"
            type="number"
            value={quantity}
            min={1}
            onChange={this.onUpdateMovie}
          />
        </div>
        <div className="pt-4 pl-4">
          <button className="btn btn-dark" type="button" onClick={this.onDelete}>
            Delete
          </button>
        </div>
      </div>
    );
  }
}

Item.propTypes = {
  onUpdate: PropTypes.func.isRequired,
  movie: PropTypes.shape({
    id: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
  }).isRequired,
  quantity: PropTypes.number.isRequired,
};
export const Component = Item;
export default Item;
