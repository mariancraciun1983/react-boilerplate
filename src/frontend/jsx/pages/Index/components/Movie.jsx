import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

import { updateMovie } from '../../../../core/stores/reducers/cart';

const actionsToProps = (dispatch) => ({
  addToCart: (id) => dispatch(updateMovie(id, 1)),
});

class Movie extends React.PureComponent {
  constructor(props) {
    super(props);
    this.addToCart = this.addToCart.bind(this);
  }

  addToCart() {
    const { addToCart, movie } = this.props;
    addToCart(movie.id);
  }

  render() {
    const { movie } = this.props;
    return (
      <div className="col col-12 col-sm-6 col-md-4 col-xl-3 pr-3 pb-3">
        <Link to={`/movie/${movie.id}/${movie.slug}`}>
          <img src={movie.image} className="img-fluid" width="100%" alt="Movie" />
        </Link>
        <div className="d-flex pt-3">
          <div className="flex-grow-1">
            <Link to={`/movie/${movie.id}/${movie.slug}`}>
              <strong>{movie.title}</strong>
            </Link>
            <br />
            <span
              className="text-danger fw-700 mr-2"
              style={{ textDecoration: 'line-through' }}
            >
              {`${movie.discounted ? movie.discounted : ''}$`}
            </span>
            <span>
              {`${movie.price}$`}
            </span>
            <br />
          </div>
          <div className="ml-auto add-cart">
            <button type="button" className="btn btn-primary" onClick={this.addToCart}>
              <i className="fas fa-cart-plus" />
            </button>
          </div>
        </div>
      </div>
    );
  }
}

Movie.propTypes = {
  addToCart: PropTypes.func.isRequired,
  movie: PropTypes.shape({
    id: PropTypes.string,
    title: PropTypes.string,
    image: PropTypes.string,
    price: PropTypes.number,
    discounted: PropTypes.number,
    slug: PropTypes.string,
  }),
};
Movie.defaultProps = {
  movie: null,
};

export const Component = Movie;

export default connect(
  null,
  actionsToProps,
)(Movie);
