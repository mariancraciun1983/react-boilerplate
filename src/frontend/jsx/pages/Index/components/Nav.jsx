import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

import { genresList } from '../../../../core/selectors';

const stateToProps = (state) => ({
  genres: genresList(state),
});
class Nav extends React.PureComponent {
  render() {
    const { genres, genre } = this.props;
    return (
      <ul className="nav flex-column">
        <li className="nav-item">
          <div className="nav-link">
            <strong>Genres</strong>
          </div>
        </li>
        {genres.map((g) => {
          const active = (genre && genre.slug === g.slug);
          const classes = classnames({
            'nav-link': true,
            active,
          });
          return (
            <li className="nav-item" key={g.slug}>
              <Link className={classes} to={`/genre/${g.slug}`}>
                {g.name}
                { active ? '*' : ''}
              </Link>
            </li>
          );
        })}
      </ul>
    );
  }
}

Nav.propTypes = {
  genres: PropTypes.arrayOf(PropTypes.object).isRequired,
  genre: PropTypes.shape({
    slug: PropTypes.string,
  }),
};
Nav.defaultProps = {
  genre: null,
};

export const Component = Nav;
export default connect(
  stateToProps,
  null,
)(Nav);
