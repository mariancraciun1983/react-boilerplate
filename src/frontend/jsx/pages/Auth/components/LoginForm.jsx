import React from 'react';
import PropTypes from 'prop-types';
import {
  Field, reduxForm, getFormValues, submit,
} from 'redux-form';
import { connect } from 'react-redux';
import { onSubmitFail, renderField } from './formUtils';

const validate = (values) => {
  const errors = {};
  if (!values.email) {
    errors.email = 'Email is required';
  } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,30}$/i.test(values.email)) {
    errors.email = 'Invalid email address';
  }
  if (!values.password) {
    errors.password = 'Password is required';
  }
  return errors;
};

const formStateToProps = (state) => ({
  formValues: getFormValues('loginForm')(state),
});
const formActionsToProps = (dispatch) => ({
  attemptSubmit: () => dispatch(submit('loginForm')),
});
class LoginForm extends React.PureComponent {
  isSubmitDisabled() {
    const { pristine, submitting, initialValues } = this.props;
    if (submitting) return true;
    if (initialValues.email && initialValues.password) return false;
    if (pristine) return true;
    return false;
  }

  render() {
    const { handleSubmit, error, submitSucceeded } = this.props;
    return (
      <div>
        {error ? <div className="alert alert-danger">{error}</div> : null}
        {submitSucceeded ? (
          <div className="alert alert-success">Login successful</div>
        ) : null}
        <form onSubmit={handleSubmit}>
          <Field
            name="email"
            label="Email"
            component={renderField}
            type="email"
            autoComplete="email"
          />
          <Field
            name="password"
            label="Password"
            component={renderField}
            type="password"
            autoComplete="current-password"
          />
          <div className=" ">
            <button
              type="submit"
              disabled={this.isSubmitDisabled()}
              className="btn btn-success btn-lg"
            >
              Login
            </button>
          </div>
        </form>
      </div>
    );
  }
}


LoginForm.propTypes = {
  pristine: PropTypes.bool.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  submitSucceeded: PropTypes.bool.isRequired,
  submitting: PropTypes.bool.isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  error: PropTypes.any,
  initialValues: PropTypes.shape({
    email: PropTypes.string,
    password: PropTypes.string,
  }),
};
LoginForm.defaultProps = {
  initialValues: { email: '', password: '' },
  error: null,
};

const LoginWithReduxForm = reduxForm({
  form: 'loginForm',
  validate,
  onSubmitFail,
})(LoginForm);
export const Component = LoginForm;
export default connect(
  formStateToProps,
  formActionsToProps,
)(LoginWithReduxForm);
