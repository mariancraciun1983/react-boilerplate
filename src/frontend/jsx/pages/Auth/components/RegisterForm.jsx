import React from 'react';
import PropTypes from 'prop-types';
import {
  Field,
  reduxForm,
  getFormValues,
  submit,
} from 'redux-form';
import { connect } from 'react-redux';
import { onSubmitFail, renderField } from './formUtils';

const validate = (values) => {
  const errors = {};
  if (!values.name) {
    errors.name = 'Name is required';
  } else if (!values.email) {
    errors.email = 'Email is required';
  } else if (!/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,30}$/i.test(values.email)) {
    errors.email = 'Invalid email address';
  }
  if (!values.password) {
    errors.password = 'Password is required';
  }
  return errors;
};


const formStateToProps = (state) => ({
  formValues: getFormValues('registerForm')(state),
});
const formActionsToProps = (dispatch) => ({
  attemptSubmit: () => dispatch(submit('registerForm')),
});
class RegisterForm extends React.PureComponent {
  isSubmitDisabled() {
    const { pristine, submitting, initialValues } = this.props;
    if (submitting) return true;
    if (initialValues.email && initialValues.password && initialValues.name) return false;
    if (pristine) return true;
    return false;
  }

  render() {
    const {
      handleSubmit,
      error,
      submitSucceeded,
    } = this.props;
    return (
      <div>
        {error ? <div className="alert alert-danger">{error}</div> : null}
        {submitSucceeded ? <div className="alert alert-success">Register successful</div> : null}
        <form onSubmit={handleSubmit}>
          <Field name="name" label="Name" component={renderField} type="name" autoComplete="name" />
          <Field
            name="email"
            label="Email"
            component={renderField}
            type="email"
            autoComplete="email"
          />
          <Field
            name="password"
            label="Password"
            component={renderField}
            type="password"
            autoComplete="current-password"
          />
          <div className=" ">
            <button
              type="submit"
              disabled={this.isSubmitDisabled()}
              className="btn btn-success btn-lg"
            >
              Register
            </button>
          </div>
        </form>
      </div>
    );
  }
}

RegisterForm.propTypes = {
  pristine: PropTypes.bool.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  submitSucceeded: PropTypes.bool.isRequired,
  submitting: PropTypes.bool.isRequired,
  // eslint-disable-next-line react/forbid-prop-types
  error: PropTypes.any,
  initialValues: PropTypes.shape({
    email: PropTypes.string,
    name: PropTypes.string,
    password: PropTypes.string,
  }),
};
RegisterForm.defaultProps = {
  initialValues: { email: '', password: '', name: '' },
  error: null,
};

export const Component = RegisterForm;
const RegisterWithReduxForm = reduxForm({
  form: 'registerForm',
  validate,
  onSubmitFail,
})(RegisterForm);
export default connect(
  formStateToProps,
  formActionsToProps,
)(RegisterWithReduxForm);
