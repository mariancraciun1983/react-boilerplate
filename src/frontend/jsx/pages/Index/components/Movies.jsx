import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import Movie from './Movie';

import { moviesByOptionalGenre } from '../utils/selectors';

const stateToProps = (state, props) => {
  const { genre } = props;
  return {
    movies: moviesByOptionalGenre(state, { genre }),
  };
};
class Movies extends React.PureComponent {
  render() {
    const { movies } = this.props;
    return (
      <div className="row">
        {movies.map((movie) => (
          <Movie key={movie.id} movie={movie} />
        ))}
      </div>
    );
  }
}

Movies.propTypes = {
  movies: PropTypes.arrayOf(PropTypes.object).isRequired,
};
export const Component = Movies;
export default connect(
  stateToProps,
  null,
)(Movies);
