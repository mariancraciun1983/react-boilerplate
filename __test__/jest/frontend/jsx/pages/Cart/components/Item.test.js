/// <reference types="jest" />
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import {Component} from '../../../../../../../src/frontend/jsx/pages/Cart/components/Item';
import mock from '../../../../../__mocks__/store';
describe("components/pages/Cart/components/Item", () => {
  let props;
  beforeEach(() => {
    props =  {
      onUpdate: jest.fn(),
      movie: mock.fullStore.movies.list[0],
      quantity: 1,
    }
  });
  it("Renders the Item", () => {
    expect.assertions(1);
    const tree = renderer
    .create(<Component {...props} />)
    .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Handles Changes", () => {
    expect.assertions(2);
    const wrapper  = shallow(<Component {...props} />)
    wrapper.find('input').at(0).simulate('change', {target: {value: '12'}})
    expect(props.onUpdate).toHaveBeenCalledWith(props.movie.id, 12);

    wrapper.find('button').at(0).simulate('click')
    expect(props.onUpdate).toHaveBeenCalledWith(props.movie.id, 0);

  });


});
