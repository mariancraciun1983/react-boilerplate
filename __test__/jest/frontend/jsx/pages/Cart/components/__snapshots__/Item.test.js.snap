// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/pages/Cart/components/Item Renders the Item 1`] = `
<div
  className="d-flex mb-3 pb-3 border-bottom"
>
  <div
    className="mr-auto d-inline-flex"
  >
    <img
      alt="Movie"
      src="/assets/images/movies/1.svg"
    />
    <div
      className="p-3"
    >
      <strong>
        Gunner Palace
      </strong>
      <br />
      <span>
        Price: 2.39$
      </span>
    </div>
  </div>
  <div
    className=""
  >
    Quantity:
    <input
      className="form-control"
      min={1}
      onChange={[Function]}
      type="number"
      value={1}
    />
  </div>
  <div
    className="pt-4 pl-4"
  >
    <button
      className="btn btn-dark"
      onClick={[Function]}
      type="button"
    >
      Delete
    </button>
  </div>
</div>
`;
